File: lib/src/mdcweb/reg_elem.dart
@JS()
library reg_elem;

import 'dart:html';
import "package:js/js.dart";

@JS()
class MdcDialogJS {
  external MdcDialogJS(Element e);
  external void open();
  external void close();
}

@JS()
class MdcSnackbarJS {
  external MdcSnackbarJS(Element e);
  external void open();
  external set labelText(String text);
}

@JS()
class MdcDrawerJS {
  external MdcDrawerJS(Element e);
  external set open(bool state);
}

@JS()
class MdcListJS {
  external MdcListJS(Element e);
}

@JS()
class MdcTextFieldJS {
  external MdcTextFieldJS(Element e);
  external String get value;
}

@JS()
class MdcSliderJS {
  external MdcSliderJS(Element e);
  external int get value;
  external void layout();//Re-computes layout
}

File: lib/js/reg_elem.js

function buildSlider(element, listenChange) {
  console.log('reg_elem.buildSlider');
  var e = new mdc.slider.MDCSlider(element);
  e.listen('MDCSlider:change', () => listenChange(e.value));
  return element;
}

function MdcDialogJS(element) {
  console.log('reg_elem.MdcDialogJS');
  return new mdc.dialog.MDCDialog(element);
}

function MdcSnackbarJS(element) {
  console.log('reg_elem.MdcSnackbarJS');
  var e = new mdc.snackbar.MDCSnackbar(element);
  e.showText = (text) => {
    const dataObj = {
      message: text
    }
    e.show(dataObj);
  }
  return e;
}

function MdcDrawerJS(element) {
  console.log('reg_elem.MdcDrawerJS');
  return new mdc.drawer.MDCDrawer(element);
}

function MdcListJS(element) {
  console.log('reg_elem.MdcListJS');
  return new mdc.list.MDCList(element);
}

function MdcTextFieldJS(element) {
  console.log('reg_elem.MdcTextFieldJS');
  return new mdc.textField.MDCTextField(element);
}
